android/src/main/java/com/rnfs/Uploader.java

```diff
@@ -117,11 +117,11 @@ private void upload(UploadParams params, UploadResult result) throws Exception {
             if (mParams.onUploadBegin != null) {
                 mParams.onUploadBegin.onUploadBegin();
             }
+            long requestLength = totalFileLength;
+            connection.setFixedLengthStreamingMode((int)requestLength);
```

Comment:
This needs to be after the `if` statement to avoid issues with non-binary only uploads.